dispatcher/nm-dispatcher.c

```diff
@@ -575,41 +575,87 @@ dispatch_one_script (Request *request)
 	return FALSE;
 }
 
-static GSList *
-find_scripts (const char *str_action)
+static int
+_compare_basenames (gconstpointer a, gconstpointer b)
+{
+	const char *basename_a = strrchr (a, '/');
+	const char *basename_b = strrchr (b, '/');
+	int ret;
+
+	g_return_val_if_fail (basename_a, 0);
+	g_return_val_if_fail (basename_b, 0);
```

Comment:
it's easy to see that this won't fail. Please use `nm_assert()`.

Also, the first `strcmp()` is not supposed to return equality. Add another `nm_assert_not_reached (); return 0;` instead of `strcmp (a, b);`.

Reply:
Ok.